selfdrive/modeld/modeld.cc

```diff
@@ -48,18 +49,13 @@ void* live_thread(void *arg) {
 
   while (!do_exit) {
     for (auto sock : poller->poll(10)){
-      Message * msg = sock->receive();
+      MessageReader amsg = sock->receive();
+      if (!amsg) continue;
 
-      auto amsg = kj::heapArray<capnp::word>((msg->getSize() / sizeof(capnp::word)) + 1);
-      memcpy(amsg.begin(), msg->getData(), msg->getSize());
-
-      capnp::FlatArrayMessageReader cmsg(amsg);
-      cereal::Event::Reader event = cmsg.getRoot<cereal::Event>();
-
-      if (event.isLiveCalibration()) {
+      if (amsg.getEvent().isLiveCalibration()) {
```

Comment:
Hmm, two calls to getEvent again, maybe
auto event = amsg.getEvent().
event.isLiveCalibration()
event.getLiveCalibration().getExtrinsicMatrix()

I don't think the compiler can do this automatically

Reply:
hmm